### PR TITLE
Feature: worker operation cancelation watcher

### PR DESCRIPTION
**Pull request by luizmiranda7**
## Description

This PR aims to create the operation cancelation watcher, it is composed by:

* A Redis pub/sub subscriber that receives the cancelation events;
* A map of context cancelation function per scheduler name and operation ID;
* Once the event is consumed, the operation manager fetches the context cancelation function and calls it in order to cancel the operation;

### CODE REVIEW

File: internal/adapters/operation_flow/redis/redis.go [hunk 1]
```diff
@@ -81,6 +84,40 @@ func (r *redisOperationFlow) ListSchedulerPendingOperationIDs(ctx context.Contex
 	return operationsIDs, nil
 }
 
+func (r *redisOperationFlow) WatchOperationCancelationRequests(ctx context.Context) chan ports.OperationCancelationRequest {
+	sub := r.client.Subscribe(ctx, watchOperationCancelationRequestKey)
+
+	resultChan := make(chan ports.OperationCancelationRequest, 10000)
```
Comment (gabrielcorado) [inline internal/adapters/operation_flow/redis/redis.go:90]: Why is this buffer so large?

File: internal/core/services/operation_manager/operation_manager.go [hunk 2]
```diff
@@ -31,9 +31,12 @@ import (
 	"github.com/topfreegames/maestro/internal/core/entities/operation"
 	"github.com/topfreegames/maestro/internal/core/operations"
 	"github.com/topfreegames/maestro/internal/core/ports"
+	"github.com/topfreegames/maestro/internal/core/ports/errors"
+	"go.uber.org/zap"
 )
 
 type OperationManager struct {
+	operationCancelFunctions        map[string]map[string]context.CancelFunc
```
Comment (GuilhermBrSp) [inline internal/core/services/operation_manager/operation_manager.go:39]: @luizmiranda7 this is a `map[schedulerId]map[operationId]context.CancelFunc`, right? WDYT about using some type alias for more readability here?

File: internal/core/workers/operation_execution_worker/operation_execution_worker.go [hunk 3]
```diff
@@ -91,20 +87,24 @@ func (w *OperationExecutionWorker) Start(ctx context.Context) error {
...
+		if !def.ShouldExecute(ctx, []*operation.Operation{}) {
+			w.evictOperation(ctx, loopLogger, op)
 			reportOperationEvicted(w.schedulerName, op.DefinitionName, LabelShouldNotExecute)
 			continue
 		}
 
 		loopLogger.Info("Starting to process operation")
 
-		err = w.operationManager.StartOperation(operationContext, op)
+		// TODO(gabrielcorado): when we introduce operation cancelation this is
```
Comment (gabrielcorado) [inline internal/core/workers/operation_execution_worker/operation_execution_worker.go:103]: I think we can remove this `TODO`.

File: internal/adapters/operation_flow/redis/redis_test.go [hunk 4]
```diff
@@ -164,3 +167,77 @@ func TestNextOperationID(t *testing.T) {
 		}, 5*time.Second, 100*time.Millisecond)
 	})
 }
+
+func TestWatchOperationCancelationRequests(t *testing.T) {
+	schedulerName := uuid.New().String()
+	operationID := uuid.New().String()
+
+	t.Run("successfully receives the scheduler name and operation ID to cancel", func(t *testing.T) {
```
Comment (GuilhermBrSp) [inline internal/adapters/operation_flow/redis/redis_test.go:175]: @luizmiranda7 WDYT about adding a test case when it fails to `Unmarshal` the incoming message? it should not return any error nor send any request to the channel, right?

**Review by gabrielcorado — APPROVED** [review]

**Review by GuilhermBrSp — APPROVED** [review]

File: internal/adapters/operation_flow/redis/redis.go (hunk 1 above)
Comment (luizmiranda7) [reply internal/adapters/operation_flow/redis/redis.go:90]: Reducing it...

File: internal/core/workers/operation_execution_worker/operation_execution_worker.go (hunk 3 above)
Comment (luizmiranda7) [reply internal/core/workers/operation_execution_worker/operation_execution_worker.go:103]: Removed

File: internal/adapters/operation_flow/redis/redis_test.go (hunk 4 above)
Comment (luizmiranda7) [reply internal/adapters/operation_flow/redis/redis_test.go:175]: Yeah, that's tricky to test... once there's no clear feedback to test

File: internal/core/services/operation_manager/operation_manager.go (hunk 2 above)
Comment (luizmiranda7) [reply internal/core/services/operation_manager/operation_manager.go]: Changed

**Pull request merged**